regret/_api.py

```diff
@@ -211,6 +212,54 @@ def __init_subclass__(Subclass, **kwargs):
 
         return deprecate
 
+    def module(
+        self,
+        version,
+        replacement=None,
+        removal_date=None,
+        addendum=None,
+    ):
+        """
+        Deprecate a module at import time.
+
+        Arguments:
+
+            version:
+
+                the first version in which the deprecated object was
+                considered deprecated
+
+            replacement:
+
+                optionally, an object that is the (direct or indirect)
+                replacement for the functionality previously performed
+                by the deprecated callable
+
+            removal_date (datetime.date):
+
+                optionally, a date when the object is expected to be
+                removed entirely
+
+            addendum (str):
+
+                an optional additional message to include at the end of
+                warnings emitted for this deprecation
+        """
+        previous_module_globals = inspect.currentframe().f_back.f_globals
+        previous_module_globals['__doc__'] = self._new_docstring(
```

Comment:
I tried to get the module object like this...but the result was a module with an empty `__doc__`.
```
module = __import__(previous_module_globals['__name__'])
```